src/events/tls.py

```diff
@@ -108,6 +109,10 @@ def _on_certificates_joined(self, event: RelationJoinedEvent) -> None:
     def _on_certificate_available(self, event: CertificateAvailableEvent) -> None:
         """Handler for `certificates_available` event after provider updates signed certs."""
         # avoid setting tls files and restarting
+        if not self.workload.container.can_connect():
```

Comment:
This is already available in `workload` object as `self.workload.ready()`.

Reply:
Thanks, I somehow missed that. Changed in 997d875ec644db8fe0a2d6bf187b6b7fa7dca8f1